test/spectests/spectests.cpp

```diff
@@ -406,40 +407,33 @@ class test_runner
         return it_instance->second.get();
     }
 
-
-    static fizzy::Value read_integer_value(const json& v)
-    {
-        assert(
-            v.at("type").get<std::string>() == "i32" || v.at("type").get<std::string>() == "i64");
-
-        // JSON tests have all values including floats serialized as 64-bit unsigned integers
-        return std::stoull(v.at("value").get<std::string>());
-    }
-
-    template <typename T>
-    static fizzy::Value read_floating_point_value(const json& v)
+    std::optional<fizzy::test::TypedValue> read_value(const json& v)
     {
-        assert(
-            v.at("type").get<std::string>() == "f32" || v.at("type").get<std::string>() == "f64");
-        assert(!is_canonical_nan(v) && !is_arithmetic_nan(v));
+        const auto type = v.at("type").get<std::string>();
 
         // JSON tests have all values including floats serialized as 64-bit unsigned integers.
-        const uint64_t uint_value = std::stoull(v.at("value").get<std::string>());
-        /// The unsigned integer type matching the size of this floating-point type.
-        using UintType = typename fizzy::test::FP<T>::UintType;
-        return fizzy::test::FP<T>{static_cast<UintType>(uint_value)}.value;
-    }
-
-    std::optional<fizzy::Value> read_value(const json& v)
-    {
-        const auto type = v.at("type").get<std::string>();
+        const uint64_t raw_value = std::stoull(v.at("value").get<std::string>());
 
-        if (type == "i32" || type == "i64")
-            return read_integer_value(v);
+        if (type == "i32")
+        {
+            assert(static_cast<uint32_t>(raw_value) == raw_value && "overflow if i32 test value");
```

Comment:
```suggestion
            assert(static_cast<uint32_t>(raw_value) == raw_value && "overflow in i32 test value");
```

Reply:
What exactly is the string part doing? It i just a true so no op. Why not leave it as a comment instead?

Reply:
It's a comment, that will be in the output when assertion fails, I think. Could be a regular commit, but I don't mind merging it like this.

Reply:
Ah right, it will be "literalized". Yeah, could keep it this way.